src/icastats_shared.c

```diff
@@ -59,41 +58,40 @@ int stats_mmap(int user)
 				    O_CREAT | O_RDWR,
 				    S_IRUSR | S_IWUSR);
 
-	if (stats_shm_handle == NOT_INITIALIZED)
+	if (stats_shm_handle == -1)
 		return -1;
 
 	if ((user > 0 && geteuid() == 0) &&
-	    (fchown(stats_shm_handle, user, user) == -1)) {
-		close(stats_shm_handle);
-		return -1;
-	}
+	    (fchown(stats_shm_handle, user, user) == -1))
+		goto err;
 
-	if (fstat(stats_shm_handle, &stat_buf)) {
-		close(stats_shm_handle);
-		return -1;
-	}
+	if (fstat(stats_shm_handle, &stat_buf))
+		goto err;
 
-	if (ftruncate(stats_shm_handle, STATS_SHM_SIZE) == -1) {
-		close(stats_shm_handle);
-		return -1;
-	}
+	if (ftruncate(stats_shm_handle, STATS_SHM_SIZE) == -1)
+		goto err;
 
 	stats = (stats_entry_t *) mmap(NULL, STATS_SHM_SIZE, PROT_READ |
 					 PROT_WRITE, MAP_SHARED,
 					 stats_shm_handle, 0);
+
 	if (stats == MAP_FAILED){
-		close(stats_shm_handle);
 		stats = NULL;
-		return -1;
+		goto err;
 	}
 
 	if (stat_buf.st_size != STATS_SHM_SIZE)
 		memset(stats, 0, STATS_SHM_SIZE);
 
+	close(stats_shm_handle);
 	return 0;
+
+err:
+	close(stats_shm_handle);
+	return -1;
```

Comment:
Use rc as return value and merge err label with good path. I will fix this in the next version of the series.